cli/tsc/dts/lib.deno.ns.d.ts

```diff
@@ -6490,7 +6484,10 @@ declare namespace Deno {
    * @category HTTP Server
    */
   export function serve(
-    options: ServeOptions | ServeTlsOptions,
+    options:
+      | ServeOptions
+      | ServeTlsOptions
+      | (ServeTlsOptions & TlsCertifiedKeyOptions),
     handler: ServeHandler,
```

Comment:
@marvinhagemeister @bartlomieju What do you guys think of this? We can make `ServeTlsOptions` an empty interface which will allow for older code using `ServeTlsOptions` to continue to typecheck.